moon/test_controller_round3.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+from unittest.mock import MagicMock
+import numpy as np
+from random import Random
+import math
+
+from shapely.geometry import LineString
+from shapely.geometry import Point
+
+try: 
+    from matplotlib import pyplot as plt
+    has_matplotlib = True
```

Comment:
I see that you draw circles based on the installation - that is not good strategy as we may need matplotlib in base OSGAR one day and the tests would hang